# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/wasm/multi-value/call-ref.js
# Commit: 920da79f43fc
# Full Hash: 920da79f43fcf5802ebd5eb2bf610a18e821dacc
# Author: Andy Wingo <wingo@igalia.com>
# Date: 2020-03-19 03:53:34
# Regressor Bug: 1620197
# File Overlap Count: 1
# Description:
#   Bug 1620197 - Enable multiple results from WebAssembly functions r=lth
#   
#   This patch enables multi-value calls and returns, adding some tests, and
#   conditionally disabling a couple expect-fail reftests that now pass.
#   
# ==============================================================================

diff -r 98d7f9e4839f -r 920da79f43fc js/src/jit-test/tests/wasm/multi-value/call-ref.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/wasm/multi-value/call-ref.js	Wed Mar 18 16:36:45 2020 +0000
@@ -0,0 +1,31 @@
+// |jit-test| skip-if: !wasmReftypesEnabled()
+
+let counter = 0;
+let i = new WebAssembly.Instance(
+    new WebAssembly.Module(wasmTextToBinary(`
+      (module
+        (func $boxNextInt (import "imports" "boxNextInt")
+          (result anyref))
+        (func $unboxInt (import "imports" "unboxInt")
+          (param anyref) (result i32))
+
+        (func $boxNextTwoInts (result anyref anyref)
+          call $boxNextInt
+          call $boxNextInt)
+
+        (func $unboxTwoInts (param anyref anyref) (result i32 i32)
+          local.get 0
+          call $unboxInt
+          local.get 1
+          call $unboxInt)
+
+        (func $addNextTwoInts (export "addNextTwoInts") (result i32)
+          call $boxNextTwoInts
+          call $unboxTwoInts
+          i32.add))`)),
+    {imports: {boxNextInt: () => ({val: counter++}),
+               unboxInt: box => box.val}});
+
+for (let n = 0; n < 200000; n += 2) {
+    assertEq(i.exports.addNextTwoInts(), n * 2 + 1);
+}